schema_based_on_diagram.sql

```diff
@@ -0,0 +1,63 @@
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP NOT NULL,
+    STATUS VARCHAR(20) NOT NULL,
+    patient_id INT NOT NULL,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(50) NOT NULL,
+    date_of_birth DATE NOT NULL,
+    PRIMARY KEY (id)
+);
+
+ALTER TABLE medical_histories
+ADD CONSTRAINT fk_patient_id FOREIGN KEY (patient_id) REFERENCES patients(id) ON DELETE CASCADE;
+
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories(patient_id);
+
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL(10,2) NOT NULL,
+  generated_at TIMESTAMP NOT NULL,
+  paid_at TIMESTAMP NOT NULL,
+  medical_history_id INT,
+  CONSTRAINT fk_medical_history_id FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id) ON DELETE CASCADE,
+  PRIMARY KEY (id)
+);
+
+CREATE INDEX idx_invoices_medical_history_id ON invoices(medical_history_id);
+
+CREATE TABLE invoice_items (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL(10,2) NOT NULL,
+  quantity INT NOT NULL,
+  total_price DECIMAL(10,2) NOT NULL,
+  invoice_id INT,
+  CONSTRAINT fk_invoice_id FOREIGN KEY (invoice_id) REFERENCES invoices(id) ON DELETE CASCADE,
+  treatment_id INT NOT NULL,
```

Comment:
- [ Required ] I think it would be nice if you add a `foreign key` constraint to the `treatment_id` column since it is supposed to be a reference of column `id` in the `treatments` table in the `invoice_items` table. Once you've added the foreign key constraint kindly remember to add an `index` to the `treatment_id` column the same way you've done for other foreign keys this will speed up search queries where the column is involved.

## Example code
```SQL
ADD CONSTRAINT fk_treatment_id FOREIGN KEY (treatment_id) REFERENCES treatments(id)

CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
```

Reply:
Hi @Felix45! thank you for reviewing this PR! 🎯

Absolutely! It seems we have overlooked it 🥲, thank you for pointing it out!

Changes will be made to apply the suggested required changes.

Cheers, and happy coding! 🤠 

@ITurres @mwismann